jenkins/PerfCI_Chaos/03_PerfCI_Choas_VMs_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,197 @@
+                    if (!env.TEMP_DIR) {
+                        error "Failed to create temporary directory!"
+                    }
+
+                    // Set file paths
+                    env.GOOGLE_DRIVE_CREDENTIALS = "${env.TEMP_DIR}/${GOOGLE_CREDENTIALS_FILE}"
+                    env.GOOGLE_DRIVE_TOKEN = "${env.TEMP_DIR}/${GOOGLE_TOKEN_FILE}"
+
+                    // Prepare credential files
+                    sh """
+                        set -e
+                        umask 77
+                        cp '${GOOGLE_DRIVE_CREDENTIALS_FILE}' '${env.GOOGLE_DRIVE_CREDENTIALS}' || error "Failed to copy ${GOOGLE_DRIVE_CREDENTIALS_FILE}"
+                        cp '${GOOGLE_DRIVE_TOKEN_FILE}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to copy ${GOOGLE_DRIVE_TOKEN_FILE}"
+                    """
+                }
+            }
+        }
+
+        stage('Fedora Deployment') {
+            steps {
+                script {
+                    sh """
+                        sudo podman run --rm -t \
+                        -e WORKLOAD='bootstorm_vm' \
+                        -e KUBEADMIN_PASSWORD='${KUBEADMIN_PASSWORD}' \
+                        -e ELASTICSEARCH='${ELASTICSEARCH}' \
+                        -e ELASTICSEARCH_PORT='${ELASTICSEARCH_PORT}' \
+                        -e IBM_REGION_NAME='${IBM_REGION_NAME}' \
+                        -e IBM_ENDPOINT_URL='${IBM_ENDPOINT_URL}' \
+                        -e IBM_ACCESS_KEY_ID='${IBM_ACCESS_KEY_ID}' \
+                        -e IBM_SECRET_ACCESS_KEY='${IBM_SECRET_ACCESS_KEY}' \
+                        -e IBM_BUCKET='${IBM_BUCKET}' \
+                        -e IBM_KEY='${IBM_KEY}' \
+                        -e GOOGLE_DRIVE_PATH='${GOOGLE_DRIVE_PATH}' \
+                        -e GOOGLE_DRIVE_CREDENTIALS_FILE='${GOOGLE_DESTINATION_PATH}/${GOOGLE_CREDENTIALS_FILE}' \
+                        -e GOOGLE_DRIVE_TOKEN_FILE='${GOOGLE_DESTINATION_PATH}/${GOOGLE_TOKEN_FILE}' \
+                        -e GOOGLE_DRIVE_SHARED_DRIVE_ID='${GOOGLE_DRIVE_SHARED_DRIVE_ID}' \
+                        -e RUN_ARTIFACTS_URL='${RUN_ARTIFACTS_URL}' \
+                        -e RUN_TYPE='${RUN_TYPE}' \
+                        -e SAVE_ARTIFACTS_LOCAL=${SAVE_ARTIFACTS_LOCAL} \
+                        -e ENABLE_PROMETHEUS_SNAPSHOT=${ENABLE_PROMETHEUS_SNAPSHOT} \
+                        -e SCALE='${BOOTSTORM_SCALE}' \
+                        -e SCALE_NODES='${SCALE_NODES}' \
+                        -e THREADS_LIMIT='${THREADS_LIMIT}' \
+                        -e DELETE_ALL=${DELETE_ALL} \
+                        -e RUN_STRATEGY=${RUN_STRATEGY} \
+                        -e TIMEOUT='${TIMEOUT}' \
+                        -e log_level='INFO' \
+                        -v '${KUBECONFIG_PATH}:${KUBECONFIG_PATH}' \
+                        -v '${TEMP_DIR}:${GOOGLE_DESTINATION_PATH}' \
+                        --privileged '${QUAY_BENCHMARK_RUNNER_REPOSITORY}'
```

Comment:
Again, do these get expanded by the shell or by Jenkins?

Reply:
All related to Jenkins environment variables and secrets

Reply:
Hard coded variable
RUN_STRATEGY = 'True'